test/controller_test.rb

```diff
@@ -282,4 +283,48 @@ class PostsControllerTest < ActionDispatch::IntegrationTest
     json = JSON.parse(@response.body)
     assert_equal [post1.id], (json.map { |post| post.fetch("id") })
   end
+
+  test "it filters on integers for array collections" do
+    Post.create!(priority: 7)
+    Post.create!(priority: 5)
+
+    get("/posts/list", params: { filters: { priority: 7 } })
+
+    json = JSON.parse(@response.body)
+
+    assert_equal [7], (json.map { |post| post["priority"] })
+  end
+
+  test "it filters on strings for array collections" do
+    Post.create!(title: "a")
+    Post.create!(title: "B")
+
+    get("/posts/list", params: { filters: { title: "B" } })
+
+    json = JSON.parse(@response.body)
+
+    assert_equal ["B"], (json.map { |post| post["title"] })
+  end
+
+  test "it filters on booleans for array collections" do
+    Post.create!(visible: true)
+    Post.create!(visible: false)
+
+    get("/posts/list", params: { filters: { visible: true } })
+
+    json = JSON.parse(@response.body)
+
+    assert_equal [true], (json.map { |post| post["visible"] })
+  end
+
+  test "it filters on decimals for array collections" do
+    Post.create!(rating: 1.2)
+    filtered_post = Post.create!(rating: 5.0)
+
+    get("/posts/list", params: { filters: { rating: 5.0 } })
+
+    json = JSON.parse(@response.body)
+
+    assert_equal [filtered_post.id], (json.map { |post| post["id"] })
+  end
```

Comment:
You may want to also add a test for filtering on ranges. I think you could do it with a decimal example:
`filters: { rating: 5.0...6.3 }`

Looks like your changes don't support ranges though? In either case it may be nice to have a spec that shows that it does/ doesn't support ranges.

Reply:
Agreed, I'll see how hard it will be to implement, but I'll get a test in either way.